core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java

```diff
@@ -12,17 +12,53 @@
 package alluxio.worker;
 
 import alluxio.Server;
+import alluxio.master.AlluxioMasterService;
 import alluxio.wire.WorkerNetAddress;
 import alluxio.worker.block.BlockWorker;
 import alluxio.worker.file.FileSystemWorker;
 
 import java.net.InetSocketAddress;
+import java.util.ServiceLoader;
+
+import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * A worker in the Alluxio system.
  */
 public interface AlluxioWorkerService extends Server {
   /**
+   * Factory for creating {@link AlluxioMasterService}.
+   */
+  @ThreadSafe
+  final class Factory {
+    /** The worker service loaders. */
+    private static ServiceLoader<WorkerFactory> sServiceLoader;
+
+    /**
+     * @return a new instance of {@link AlluxioWorkerService}
+     */
+    public static AlluxioWorkerService create() {
+      return new DefaultAlluxioWorker();
+    }
+
+    /**
+     * @return the (cached) worker service loader
+     */
+    static synchronized ServiceLoader<WorkerFactory> getServiceLoader() {
```

Comment:
This method seems like an implementation detail, not something that a user of the class cares about. Would it make more sense to put it in the concrete class?